### PR TITLE
7 post comment

### PR DESCRIPTION
**Pull request by kimberleyfisherx**
Added API endpoint for commenting on articles

### CODE REVIEW

File: api/__tests__/all.test.js [hunk 1]
```diff
@@ -180,3 +180,58 @@ test("comments sorted by most recent first", () => {
...
+      .then((response) => {
+        const commentObject = response.body.comment;
+        const expectedCommentObject = {
+          author: "lurker",
+          body: "weird article",
+        };
+
+        expect(commentObject).toMatchObject(expectedCommentObject);
+      });
+  });
```
Comment (B7Bprog) [inline api/__tests__/all.test.js:184]: Please add another test for a successful response to check if extra properties in the post object are ignored. for example when posting with this:
`{ username: "lurker", body: "weird article", extraProp: "extraValue" }`

File: api/__tests__/all.test.js [hunk 2]
```diff
@@ -180,3 +180,58 @@ test("comments sorted by most recent first", () => {
...
+  test("if comment only contains white space returns correct error message", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({ username: "lurker", body: " " }) //comment is just a space
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toEqual("Please enter a comment");
+      });
+  });
+  describe("POST /api/articles/:article_id/comments", () => {
```
Comment (B7Bprog) [inline api/__tests__/all.test.js]: This doesn't need a separate describe block here as it's for the same endpoint.

File: api/__tests__/all.test.js [hunk 3]
```diff
@@ -180,3 +180,58 @@ test("comments sorted by most recent first", () => {
...
+  });
+  test("if comment only contains white space returns correct error message", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({ username: "lurker", body: " " }) //comment is just a space
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toEqual("Please enter a comment");
+      });
+  });
```
Comment (B7Bprog) [inline api/__tests__/all.test.js:211]: Please add another test where the username is valid, but not in the database. (404)

File: api/models/models.api.js [hunk 4]
```diff
@@ -66,3 +66,34 @@ exports.getCommentsById = (id) => {
...
+exports.postComment = (id, comment) => {
+  const articleIdNum = id.article_id;
+  const commentBody = comment.body;
+  const commentAuthor = comment.username;
+
+  if (!commentBody || commentBody.trim() === "") {
+    return Promise.reject({ status: 400, msg: "Please enter a comment" }); // this looks for an empty comment or comment with only a space
+  }
+
+  if (!commentAuthor) {
```
Comment (B7Bprog) [inline api/models/models.api.js]: Would psql throw an error for this?

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Kimberly, please check my comments and refactor your code, then please send this PR back to us with the updates.

File: api/__tests__/all.test.js [hunk 5]
```diff
@@ -180,3 +166,85 @@ test("comments sorted by most recent first", () => {
...
+  test("Extra properties in post object are ignored", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({
+        username: "lurker",
+        body: "weird article",
+        extraProp: "extraValue",
+      })
```
Comment (Jodiech) [inline api/__tests__/all.test.js:228]: Nice test for ignoring additional properties 👍 I'd consider keeping your 201 tests together to keep your code organised!

**Review by Jodiech — APPROVED** [review]
Great work on amending your test suite as per Bélas comments. Happy for you to merge! 😄

**Pull request merged**